pkg/resources/cruisecontrol/topicManager.go

```diff
@@ -19,73 +19,73 @@ import (
 	"crypto/tls"
 	"crypto/x509"
 	"fmt"
-	"time"
 
+	"emperror.dev/errors"
 	banzaicloudv1alpha1 "github.com/banzaicloud/kafka-operator/pkg/apis/banzaicloud/v1alpha1"
 	"github.com/banzaicloud/kafka-operator/pkg/resources/kafka"
+	"github.com/go-logr/logr"
 	"github.com/goph/emperror"
-	kafkaGo "github.com/segmentio/kafka-go"
+
+	"github.com/Shopify/sarama"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func generateCCTopic(cluster *banzaicloudv1alpha1.KafkaCluster, client client.Client) error {
+func generateCCTopic(cluster *banzaicloudv1alpha1.KafkaCluster, client client.Client, log logr.Logger) error {
 
-	dialer := &kafkaGo.Dialer{}
+	conf := sarama.NewConfig()
+	conf.Version = sarama.V2_1_0_0
 
 	if cluster.Spec.ListenersConfig.SSLSecrets != nil {
 		tlsKeys := &corev1.Secret{}
 		err := client.Get(context.TODO(), types.NamespacedName{Namespace: cluster.Namespace, Name: cluster.Spec.ListenersConfig.SSLSecrets.TLSSecretName}, tlsKeys)
 		if err != nil {
 			return emperror.Wrap(err, "could not get TLS secret to create CC topic")
 		}
-		serverCert := tlsKeys.Data["serverCert"]
-		serverKey := tlsKeys.Data["serverKey"]
+		clientCert := tlsKeys.Data["clientCert"]
+		clientKey := tlsKeys.Data["clientKey"]
 		caCert := tlsKeys.Data["caCert"]
-		x509ServerCert, err := tls.X509KeyPair(serverCert, serverKey)
+
+		x509ClientCert, err := tls.X509KeyPair(clientCert, clientKey)
 		if err != nil {
 			return err
 		}
+
 		rootCAs := x509.NewCertPool()
 		rootCAs.AppendCertsFromPEM(caCert)
 
-		dialer = &kafkaGo.Dialer{
-			Timeout:   10 * time.Second,
-			DualStack: true,
-			TLS: &tls.Config{
-				Certificates: []tls.Certificate{x509ServerCert},
-				RootCAs:      rootCAs,
-			},
-		}
-	} else {
-		dialer = &kafkaGo.Dialer{
-			Timeout:   10 * time.Second,
-			DualStack: true,
+		t := &tls.Config{
+			Certificates: []tls.Certificate{x509ClientCert},
+			RootCAs:      rootCAs,
 		}
+		conf.Net.TLS.Enable = true
+		conf.Net.TLS.Config = t
 	}
-	conn, err := dialer.Dial("tcp",
-		generateKafkaAddress(cluster))
+
+	adminClient, err := sarama.NewClusterAdmin([]string{generateKafkaAddress(cluster)}, conf)
 	if err != nil {
-		return emperror.Wrap(err, "could not create topic for CC because kafka is unavailable")
+		return emperror.Wrap(err, "Error while creating admin client to create CC topic")
 	}
-	defer conn.Close()
-	tConfig := kafkaGo.TopicConfig{
-		Topic:             "__CruiseControlMetrics",
+	defer func() { _ = adminClient.Close() }()
+
+	err = adminClient.CreateTopic("__CruiseControlMetrics", &sarama.TopicDetail{
 		NumPartitions:     12,
 		ReplicationFactor: 3,
-	}
+	}, false)
 
-	err = conn.CreateTopics(tConfig)
-	if err != nil {
-		return emperror.Wrap(err, "could not create topic for CC")
+	var tError *sarama.TopicError
+
+	if err != nil && !(errors.As(err, &tError) && tError.Err == sarama.ErrTopicAlreadyExists) {
```

Comment:
tError is already a pointer is `&` necessary here?

Reply:
yes, because the error interface is implemented by using *TopicError as receiver.
The As method dereference the pointer so if I use just `tError` here it means the type will be TopicError which actually does not implement the error interface. It is a little bit odd I know:)